staging/operator-lifecycle-manager/pkg/controller/registry/reconciler/grpc.go

```diff
@@ -192,21 +192,35 @@ func (c *GrpcRegistryReconciler) EnsureRegistryServer(catalogSource *v1alpha1.Ca
 	// recreate the pod if no existing pod is serving the latest image
 	overwritePod := overwrite || len(c.currentPodsWithCorrectImage(source)) == 0
 
-	//TODO: if any of these error out, we should write a status back (possibly set RegistryServiceStatus to nil so they get recreated)
 	sa, err := c.ensureSA(source)
 	if err != nil && !k8serror.IsAlreadyExists(err) {
+		// TODO(anik120): introduce this condition reason in the api repo
```

Comment:
We'll need to run `make vendor` to propagate any changes made to the Go packages in the relevant staging/* repositories.

Reply:
Right, introduced this here as a trailer for what the plan is and to get the plan reviewed/approved. Will `make vendor` the api repo once the idea is approved.

Reply:
Fair enough, but we build our binaries/images from the root vendor tree so it's possible that these changes won't get propagated to the CI feedback.

Reply:
We don't have a concrete downstream-specific strategy yet, namely because we don't have a mechanism that verifies that patches introduced downstream will be retained when pulling in upstream changes. In general, though, the same workflow would apply to any PR that changes the staging package code, is that we need to run `make vendor` so those changes propagate to the root vendor/ (which is why we check for unstaged changes in the verification check).

Reply:
@anik120 The order of operations doesn't matter too much here as regardless of pulling in changes made upstream, or introducing downstream-only changes, we pull Go package changes to their corresponding staging file path, and re-generate the root vendor tree. We have setup the [root modules to use the local filesystem path](https://github.com/openshift/operator-framework-olm/blob/master/go.mod#L58-L61):

```
	// use staged repositories
	github.com/operator-framework/api => ./staging/api
	github.com/operator-framework/operator-lifecycle-manager => ./staging/operator-lifecycle-manager
	github.com/operator-framework/operator-registry => ./staging/operator-registry
```

And introduced a [`cmds` package at the root](https://github.com/openshift/operator-framework-olm/blob/master/cmds.go) so the package imports are correctly remapped, without having to do that remapping ourselves:

```go
package cmds

import (
	_ "github.com/operator-framework/operator-lifecycle-manager/cmd/catalog"
	_ "github.com/operator-framework/operator-lifecycle-manager/cmd/olm"
	_ "github.com/operator-framework/operator-lifecycle-manager/cmd/package-server"

	_ "github.com/operator-framework/operator-registry/cmd/configmap-server"
	_ "github.com/operator-framework/operator-registry/cmd/initializer"
	_ "github.com/operator-framework/operator-registry/cmd/opm"
	_ "github.com/operator-framework/operator-registry/cmd/registry-server"
)
```

Reply:
And re-reading my last comment, we're technically double vendoring the OLM/API/registry packages, both in the root vendor (which is what we built from), and the staging/* repositories. That pattern is a bit strange, and it was mainly to get an MVP working sooner rather than later with the looming 4.8 feature freeze deadline, so we updated that ci/prow/verify check to ensure that when we run `make vendor` we have no unstaged changes, so we don't run into situations where the root vendor and staging/* packages are out of sync.

Reply:
Gotcha, pushed changes from `make vendor` in with the commit.